controller/src/main/java/org/jboss/as/controller/operations/global/ReadResourceHandler.java

```diff
@@ -500,6 +505,9 @@ public void execute(OperationContext context, ModelNode operation) throws Operat
                 ModelNode value = ar.response;
                 if (!value.has(FAILURE_DESCRIPTION)) {
                     sortedAttributes.put(entry.getKey(), value.get(RESULT));
+                    if (value.hasDefined(RESPONSE_HEADERS, WARNINGS)) {
+                        context.addResponseWarning(Level.WARNING, value.get(RESPONSE_HEADERS).get(WARNINGS));
+                    }
```

Comment:
This piece is still hardcoding the Level.WARNING and is not iterating over the potential list. 

If you look at how `context.addResponseWarning()` is done, it is not prepared to add a list of warnings, it adds a single warning per call.